### PR TITLE
[Naitei] Login_Google

### PR DESCRIPTION
**Pull request by ThaiKhacDuoc**
## Related Tickets
- ticket redmine

## WHAT (optional)
- Change number items `completed/total` in admin page.

## HOW
- I edit js file, inject not_vary_normal items in calculate function.

## WHY (optional)
- Because in previous version - number just depends on `normal` items. But in new version, we have `state` and `confirm_state` depends on both `normal` + `not_normal` items.

## Evidence (Screenshot or Video)

https://github.com/user-attachments/assets/831d2aab-1d3d-4f16-a63f-d7dbab7cc415

### CODE REVIEW

File: app/views/devise/mailer/email_changed.html.erb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<p>Hello <%= @email %>!</p>
```
Comment (TranMinhHuy2003) [inline app/views/devise/mailer/email_changed.html.erb]: thiếu i18n

File: app/views/devise/mailer/password_change.html.erb [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>We're contacting you to notify you that your password has been changed.</p>
```
Comment (TranMinhHuy2003) [inline app/views/devise/mailer/password_change.html.erb]: thiếu i18n

File: app/views/devise/mailer/reset_password_instructions.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<p>Hello <%= @resource.email %>!</p>
+
+<p>Someone has requested a link to change your password. You can do this through the link below.</p>
+
+<p><%= link_to 'Change my password', edit_password_url(@resource, reset_password_token: @token) %></p>
+
+<p>If you didn't request this, please ignore this email.</p>
+<p>Your password won't change until you access the link above and create a new one.</p>
```
Comment (TranMinhHuy2003) [inline app/views/devise/mailer/reset_password_instructions.html.erb]: đây nữa

**Review by TranMinhHuy2003 — COMMENTED** [review]
ông review kĩ lại nha, còn nhiều chỗ thiếu i18n á

**Comment by ThaiKhacDuoc** [comment]
ready

File: app/models/user.rb [hunk 4]
```diff
@@ -43,6 +43,16 @@ class User < ApplicationRecord
   has_one_attached :avatar
   scope :all_users, ->{all}
 
+  def self.from_omniauth access_token
+    data = access_token.info
+    user = User.where(email: data["email"]).first
+    user || User.create(name: data["name"],
```
Comment (ghost) [inline app/models/user.rb]: Dùng hàm first_or_create đi em

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ThaiKhacDuoc** [comment]
ready

**Comment by ThaiKhacDuoc** [comment]
ready

**Review by ghost — APPROVED** [review]

**Pull request merged**